macaroonbakery/macaroon.py

```diff
@@ -229,14 +277,27 @@ def third_party_caveats(self):
         '''
         return self._macaroon.third_party_caveats()
 
+    def copy(self):
+        ''' Returns a copy of the macaroon. Note that the the new
+        macaroon's namespace still points to the same underlying Namespace -
+        copying the macaroon does not make a copy of the namespace.
+        :return a Macaroon
+        '''
+        m1 = Macaroon(None, None, version=self._version,
+                      namespace=self._namespace)
+        m1._macaroon = self._macaroon.copy()
+        self._caveat_id_prefix = bytearray()
```

Comment:
This doesn't seem right - the caveat id prefix shoudn't be modified in the original, and it should be duplicated in the copy AFAICS.